<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<HEAD>
<TITLE>DNA Sequence Alignment</TITLE>
<META NAME="generator" CONTENT="me">
<META NAME="keywords" CONTENT="DNA, sequence, slignment, algorithms, survey">
<META NAME="description" CONTENT="This page contains information about the
DNA sequence alignment project done by Eric Indiran, Matt Maxel, and
Michael Dipperstein for a class on Indexing in Multimedia Databases">
</HEAD>

<HTML>
<BODY>

<H1>DNA Sequence Alignment Project</H1>

<H2>Description of Project</H2>
<P>
For this project, we will be studying and comparing techniques implemented
to align protein sequences and perform similarity comparisons.  For this
study we will be concentrating on DNA sequences, since they present the same
issues as generic protein sequences, but have data sets which are more
readily available.  We plan to focus our studies on database applications,
though a large portion of the study will be focused on the alignment of any
two DNA sequences independent of a database.  As part of our studies we hope
to analyze the following:
</P>

<UL>
<LI>Global alignment vs local alignment</LI>
<LI>Gapped alignment vs ungapped alignment</LI>
<LI>Techniques for improving comparison time by storing precomputations</LI>
<LI>Metrics used:</LI>
<UL>
<LI>The ideas behind some of the more common metrics</LI>
<LI>The effect a metric may have on the search results</LI>
</UL>
<LI>A general survey of the DNA sequence alignment databases currently
available on the Web for researchers to use</LI>
</UL>

<P>
This is a very ambitious effort for which whole graduate courses are
devoted to.  It is our intent to narrow the focus as we continue to
study the topic.  Otherwise we will not be able to finish the project
before the end of the quarter, and the material we gather will take too
long to present to the class.
</P>

<H2>Description of Datasets</H2>
<P>
Currently there are several databases actively on-line which provide DNA
string comparisons against their datasets.  To the best extent possible
we will be using the datasets that these data bases have.  In addition
there have been a few attempts at making small benchmarking datasets
which we may also use.  Where neither of these datasets will achieve our
goal, we will create our own datasets which may include a combination of
specifically tailored DNA sequences and randomly generated sequences.
</P>

<H2>Measure of Success of the Project</H2>
<P>
A successful project will be indicated by a paper which presents the
problem we studied and details the analysis of at least 3 of the areas
above (or new areas that have been discovered as we continue to study
the topic).  In addition to the paper a successful project should
include a presentation to the class.
</P>

<P>
An excellent project would demonstrate that we have studied the topic
and developed an understanding of the material.  Our write-up
and presentation may be used to demonstrate this and should include
calculated and/or empirical analysis of areas studied.
</P>

<P>
A satisfactory project would still include the study of three areas,
though it would include a cursory and purely qualitative comparison.
</P>

<H2>Timetable</H2>
<P>
At this point depending on what we find in the readings, we have two
possible timetables.
<P>

<BLOCKQUOTE>
Week 1 Cursory review of the reading materials<BR>
Week 2 Narrow focus of the project<BR>
Week 3 Detailed analysis of first topic<BR>
Week 4 Detailed analysis of second topic<BR>
Week 5 Detailed analysis of third topic<BR>
Week 6 Write-up<BR>
Week 7 Presentation<BR>
</BLOCKQUOTE>

<P>
or possibly
</P>

<BLOCKQUOTE>
Week 1 Cursory review of the reading materials<BR>
Week 2 Narrow focus of the project and distribute responsibilities<BR>
Week 3 Individual detailed analysis of assigned topic<BR>
Week 4 Continued individual analysis of topics<BR>
Week 5 Integration of topics studied<BR>
Week 6 Write-up<BR>
Week 7 Presentation<BR>
</BLOCKQUOTE>


<P>
The following is a list of papers and other material that we are
currently reviewing:
</P>

<P>
Torbjorn Rognes and Erling Seeberg<BR>
SALSA: Improved Protein Database Searching by a New Algorithm for
Assembly of Sequence Fragments into Gapped Alignments
Bioinformatics, Vol. 14 no. 10, Pages 839-845, 1998<BR><BR>

Jean-Stephen Varre, Jean-Paul Delahaye, and Eric Rivals<BR>
Transformation Distances: A Family of Dissimilarity Measures Based on
Movements of Segments<BR>
Bioinformatics, Vol. 15 no. 3, Pages 194-202, 1999<BR><BR>

Burkhard Morgenstern, Kornelie Frech, Andreas Dress, and Thomas Werner<BR>
DIALIGN: Finding Local Similarities by Multiple Sequence Alignment<BR>
Bioinformatics, Vol. 14 no. 3, Pages 290-294, 1998<BR><BR>

C. Miller, J. Gurd, and A. Brass<BR>
A RAPID Algorithm for Sequence Database Comparisons: Application to the
Identification of Vector Contamination in EMBL Databases<BR>
Bioinformatics, Vol. 15 no. 2, Pages 111-121, 1999<BR><BR>

Udo Tonges, Soren W. Perrey, Jens Stoye, and Andreas W.M. Dress<BR>
A General Method for Fast Multiple Sequence Alignment<BR>
Gene 172, 33-44, 1996<BR><BR>

Paul Hardy and Michael S. Waterman<BR>
The Sequence Alignment Software Library at USC<BR>
Version 2.0, March 1997<BR>
<A HREF="http://www-hto.usc.edu/seqaln">http://www-hto.usc.edu/seqaln</A>
<BR><BR>

K. Quandt<BR>
Software for the Analysis of DNA Sequence Elements of Transcription<BR>
January 1997<BR>
<A HREF="http://www.gsf.de/biodv/review/">http://www.gsf.de/biodv/review/</A>
<BR><BR>

Thomas Werner<BR>
The Biology and Bioinformatics of Regulatory Regions in Genomes<BR>
<A HREF="ftp://ariane.gsf.de/pub/ISMB99/tutorial_handout.pdf">
ftp://ariane.gsf.de/pub/ISMB99/tutorial_handout.pdf</A><BR><BR>

Dr. David Mount and Dr. Samuel Ward<BR>
MCB 416/516 BIOINFORMATICS AND GENOMIC ANALYSIS<BR>
Class Notes, Spring 1998<BR>
<A HREF="http://www.blc.arizona.edu/courses/wardbioinf/">
http://www.blc.arizona.edu/courses/wardbioinf/</A><BR><BR>

L. Allison, D.Powell & T.I.Dix,<BR>
Compression and Approximate Matching,<BR>
The Computer Journal, Volume 42, Issue 1, Pages 1-10, 1999<BR><BR>

L. Allison,C.S. Wallace and C.N.Yee<BR>
Inductive Inference over Macro-Molecules<BR>
TechnicalReport 90/148, Department of Computer Science, Monash
University, Austrailia 3168<BR><BR>

L. Allison<BR>
Information-Theoretic Sequence Alignment<BR>
Technical Report 98/14, School of Computer Science and Software
Engineering, Monash University,Australia 3168<BR><BR>

Dennis A. Benson, Mark S. Boguski, David J. Lipman, James Ostell and
B.F. Francis Ouellette<BR>
GenBank<BR>
Nucleic Acids Research, Vol 26, Issue 1.<BR><BR>

C.Harger, M. Skupski, J.Bingham, A. Farmer, S. Hoisie, P. Hraber, D.
Kiphart, L. Krakowski, M. McLeod, J. Schwertfeger, G. Seluja, A. Siepel,
G. Singh, D. Stamper, P.Steadman, N. Thayer, R. Thompson, P. Wargo, M.
Waugh, J.J. Zhuang and P.A. Schad<BR>
The Genome Sequence DataBase (GSDB):  Improving Data Quality and
Data Access<BR>
Nucleic Acids Research, Vol 26, Issue 1.<BR><BR>

Gene Myers<BR>
A Fast Bit-Vector Algoithm for Approximate String Matching Based on
Dynamic Programming<BR>
Ninth Combinatorial Pattern Matching Conference<BR>
accepted for publication<BR><BR>

Eugene W. Myers, Paulo Oliva and Katia Guimaraes<BR>
Reporting Exact and Approximate Regular Expression Matches<BR>
Ninth Combinatorial Pattern Matching Conference<BR>
accepted for publication.<BR><BR>

Eric L. Anson and Eugene W. Myers<BR>
ReAligner:  A Program for Refinging DNA Sequence Multi-Alignments<BR>
Journal of Computational Biology 4, 3 (1997), 369-383.<BR><BR>

Gerhard Mehldau and Gene Myers<BR>
A System for Pattern Matching Applications on Biosequences<BR>
CABIOS 9, 3 (1993), 299-314<BR><BR>

David Loewenstern, Haym Hirsh, Peter Yianilos and Michiel Noordewier<BR>
DNA Sequence Classification Using Compression-Based Induction<BR>
DIMACS Technical Report 95-04, Department of Computer Science,<BR>
Rutgers University, New Brunswick, NJ 08903, USA<BR><BR>

Burkhard Morgenstern, Angreas Dress and Thomas Werner<BR>
Multiple DNA and Protein Sequence Alignment based on
Segment-to-Segment Comparison<BR>
Proc. Natl. Acad.Sci. USA, Vol. 93, pp. 12098-12103, October 1996<BR>
Applied Mathematics<BR><BR>

Felix Wu and Guy Blelloch<BR>
Algorithms in the Real World<BR>
Lecture #20 (Pattern Matching 2)<BR><BR>

Steven E. Brenner, Cyrus Chothia and Tim J. Hubbard<BR>
Assessing Sequence Comparison Methods with Reliable Structurally
Identified Distant Evolutionary Relationships<BR>
Proc. Natl. Acad. Sci. USA  Vol. 95, pp. 6073-6078, May 1998<BR>
Biochemistry<BR><BR>

Tatiana A. Tatusova and Thomas L. Madden<BR>
BLAST 2 Sequences, a new Tool for Comparing Protein and Nucleotide
Sequences<BR>
FEMS Microbiology Letters 174 (1999) 247-250<BR><BR>

R.Irie,S. Hiraoka,N. Kasahara and K. Nagai<BR>
DNA Sequence Comparison Considering Both Amino Acid and Nucleotide
Insertions/Deletions Because of Evolution and Experimental Error<BR>
Journal of Biotechnology 69 (1999) 19-26<BR><BR>

John C. Wooton<BR>
Evaluating the Effectiveness of Sequence Analysis Algorithms Using
Measure of Relevant Information<BR>
Computers Chem.  Vol. 21, No. 4, pp. 191-202, 1997<BR><BR>

Thomas D. Schneider and David N. Mastronarde<BR>
Fast Multiple Alignment of Ungapped DNA Sequences using Information
Theory and a Relaxation Method<BR>
Discrete Applied Mathematics 71 (1996) 259-268<BR><BR>

Ron Shamir<BR>
Pairwise Alignment<BR>
Algorithms for  Molecular Biology, Lecture 2, 1998<BR><BR>

Ron Shamir<BR>
Introduction - Sequence Alignment Heuristics<BR>
Algorithms for  Molecular Biology, Lecture 3, 1998<BR><BR>

Ron Shamir<BR>
Approximatin Algorithms for Multiple Sequence Alignment<BR>
Algorithms for  Molecular Biology, Lecture 5, 1999<BR><BR>
</P>

</BODY>
</HTML>
