<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<HEAD>
<TITLE>Glossary</TITLE>
<META name="generator" content="NoteTab 4.6a">
<META NAME="keywords" CONTENT="molecular biology biochemistry glossary">
</HEAD>

<BODY TEXT="#000000" BGCOLOR="#CCCCCC" LINK="#0000CC" VLINK="#660066"
ALINK="#660066">

<H1>Glossary</H1>

<P>The majority of this document was taken from:
<A HREF="http://www.gdb.org/Dan/DOE/intro.html">
Primer on Molecular Genetics, DOE Human Genome Program,
U.S. Department of Eneregy, Office of Energy Research, June, 1992
</A></P>

<B><P>
<A NAME="adenine">
<A NAME="a">
Adenine (A):</B> A nitrogenous base, one member of the <I>base pair</I> A-T
(<I>adenine</I>-<I>thymine</I>).</P>

<B><P>
<A NAME="alleles">
Alleles:</B> Alternative forms of a genetic <I>locus</I>; a single allele for
each locus is inherited separately from each parent (e.g., at a locus for eye
color the allele might result in blue or brown eyes).</P>

<B><P>
<A NAME="amino acid">
Amino acid:</B> Any of a class of 20 molecules that are combined to form
<I>proteins</I> in living things. The sequence of amino acids in a protein and
hence protein function are determined by the <I>genetic code</I>.</P>

<B><P>
<A NAME="amplification">
Amplification:</B> An increase in the number of copies of a specific DNA
fragment; can be in vivo or in vitro. See <I>cloning, polymerase chain
reaction</I>.</P>

<B><P>
<A NAME="arrayed library">
Arrayed library:</B> Individual primary recombinant clones (hosted in <I>
phage, cosmid, YAC</I>, or other <I>vector</I>) that are placed in
two-dimensional arrays in microtiter dishes. Each primary clone can be
identified by the identity of the plate and the clone location (row and
column) on that plate. Arrayed libraries of clones can be used for many
applications, including screening for a specific gene or genomic region of
interest as well as for <I>physical mapping</I>. Information gathered on
individual clones from various genetic <I>linkage</I> and <I>physical map</I>
analyses is entered into a relational database and used to construct physical
and genetic <I>linkage maps</I> simultaneously; clone identifiers serve to
interrelate the multilevel maps. Compare <I>library, genomic library</I>.</P>

<B><P>
<A NAME="autoradiography">
Autoradiography:</B> A technique that uses X-ray film to visualize
radioactively labeled molecules or fragments of molecules; used in analyzing
length and number of DNA fragments after they are separated by gel
<I>electrophoresis</I>.</P>

<B><P>
<A NAME="autosome">
Autosome:</B> A <I>chromosome</I> not involved in sex determination. The
<I>diploid</I> human <I>genome</I> consists of 46 chromosomes, 22 pairs of
autosomes, and 1 pair of <I>sex chromosomes</I> (the X and Y chromosomes).</P>

<B><P>
<A NAME="bacteriophage">
Bacteriophage:</B> See <I>phage</I>.</P>

<B><P>
<A NAME="base pair">
<A NAME="bp">
Base pair (bp):</B> Two nitrogenous bases (<I>adenine</I> and <I>thymine</I>
or <I>guanine</I> and <I>cytosine</I>) held together by weak bonds. Two
strands of DNA are held together in the shape of a double helix by the bonds
between base pairs.</P>

<B><P>
<A NAME="base sequence">
Base sequence:</B> The order of <I>nucleotide</I> bases in a DNA molecule.</P>

<B><P>
<A NAME="base sequence analysis">
Base sequence analysis:</B> A method, sometimes automated, for determining the
<I>base sequence</I>.</P>

<B><P>
<A NAME="biotechnology">
Biotechnology:</B> A set of biological techniques developed through basic
research and now applied to research and product development. In particular,
the use by industry of <I>recombinant DNA</I>, cell fusion, and new
bioprocessing techniques.</P>

<B><P>
<A NAME="blosum">
BLOSUM:</B> Blocks Substitution Matrices for Protein Sequence Comparisons.
A group of weighted scoring matricies for alignment of <I>nucleotied</I> or
<I>DNA</I> sequences.</P>

<B><P>
<A NAME="bp">
bp</B> or <B>bps:</B> See <I>base pair</I>.</P>

<B><P>
<A NAME="cdna">
cDNA:</B> See <I>complementary DNA</I>.</P>

<B><P>
<A NAME="centimorgan">
<A NAME="cm">
Centimorgan (cM):</B> A unit of measure of <I>recombination</I> frequency. One
centimorgan is equal to a 1% chance that a marker at one genetic <I>locus</I>
will be separated from a marker at a second locus due to <I>crossing over</I>
in a single generation. In human beings, 1 centimorgan is equivalent, on
average, to 1 million <I>base pairs</I>.</P>

<B><P>
<A NAME="centromere">
Centromere:</B> A specialized <I>chromosome</I> region to which spindle fibers
attach during cell division.</P>

<B><P>
<A NAME="chromosomes">
Chromosomes:</B> The self-
replicating genetic structures of cells containing the cellular DNA that bears
in its <I>nucleotide</I> sequence the linear array of <I>genes</I>. In
<I>prokaryotes</I>, chromosomal DNA is circular, and the entire genome is
carried on one chromosome. <I>Eukaryotic</I> genomes consist of a number of
chromosomes whose DNA is associated with different kinds of <I>proteins</I>.
</P>

<B><P>
<A NAME="clone bank">
Clone bank:</B> See <I>genomic library</I>.</P>

<B><P>
<A NAME="clones">
Clones:</B> A group of cells derived from a single ancestor.</P>

<B><P>
<A NAME="cloning">
Cloning:</B> The process of asexually producing a group of cells (clones), all
genetically identical, from a single ancestor. In <I>recombinant DNA
technology</I>, the use of DNA manipulation procedures to produce multiple
copies of a single <I>gene</I> or segment of DNA is referred to as cloning
DNA.</P>

<B><P>
<A NAME="cloning vector">
Cloning vector:</B> DNA molecule originating from a <I>virus</I>, a
<I>plasmid</I>, or the cell of a higher organism into which another DNA
fragment of appropriate size can be integrated without loss of the vectors
capacity for self-replication; vectors introduce foreign DNA into host cells,
where it can be reproduced in large quantities. Examples are <I>plasmids,
cosmids</I>, and <I>yeast artificial chromosomes</I>; vectors are often
<I>recombinant</I> molecules containing DNA sequences from several sources.
</P>

<B><P>
<A NAME="cm">
cM:</B> See <I>centimorgan</I>.</P>

<B><P>
<A NAME="code">
Code:</B> See <I>genetic code</I>.</P>

<B><P>
<A NAME="codon">
Codon:</B> See <I>genetic code</I>.</P>

<B><P>
<A NAME="complementary dna">
<A NAME="cdna">
Complementary DNA (cDNA):</B> DNA that is synthesized from a <I>messenger
RNA</I> template; the single-stranded form is often used as a <I>probe</I>
in <I>physical mapping</I>.</P>

<B><P>
<A NAME="complementary sequences">
Complementary sequences:</B> <I>Nucleic acid base sequences</I> that can form
a double-stranded structure by matching <I>base pairs</I>; the complementary
sequence to G-T-A-C is C-A-T-G.</P>

<B><P>
<A NAME="conserved sequence">
Conserved sequence:</B> A <I>base sequence</I> in a DNA molecule (or an
<I>amino acid</I> sequence in a <I>protein</I>) that has remained essentially
unchanged throughout evolution.</P>

<B><P>
<A NAME="contig map">
Contig map:</B> A map depicting the relative order of a linked <I>library</I>
of small overlapping clones representing a complete chromosomal segment.</P>

<B><P>
<A NAME="contigs">
Contigs:</B> Groups of <I>clones</I> representing overlapping regions of a
<I>genome</I>.</P>

<B><P>
<A NAME="cosmid">
Cosmid:</B> Artificially constructed <I>cloning vector</I> containing the
<I>cos</I> gene of <I>phage</I> lambda. Cosmids can be packaged in lambda
phage particles for infection into <I>E. coli</I>; this permits cloning of
larger DNA fragments (up to 45 kb) than can be introduced into bacterial hosts
in <I>plasmid</I> vectors.</P>

<B><P>
<A NAME="crossing over">
Crossing over:</B> The breaking during <I>meiosis</I> of one maternal and one
paternal <I>chromosome</I>, the exchange of corresponding sections of DNA, and
the rejoining of the chromosomes. This process can result in an exchange of
alleles between chromosomes. Compare <I>recombination</I>.</P>

<B><P>
<A NAME="cytosine">
<A NAME="c">
Cytosine (C):</B> A <I>nitrogenous base</I>, one member of the <I>base pair
</I> G-C (<I>guanine</I> and <I>cytosine</I>).</P>

<B><P>
<A NAME="deoxyribonucleotide">
Deoxyribonucleotide:</B> See <I>nucleotide</I>.</P>

<B><P>
<A NAME="diploid">
Diploid:</B> A full set of genetic material, consisting of paired
<I>chromosomes</I> one chromosome from each parental set. Most animal cells
except the <I>gametes</I> have a diploid set of chromosomes. The diploid human
<I>genome</I> has 46 chromosomes. Compare <I>haploid</I>.</P>

<B><P>
<A NAME="dna">
<A NAME="deoxyribonucleic acid">
DNA (deoxyribonucleic acid):</B> The molecule that encodes genetic
information. DNA is a double-stranded molecule held together by weak bonds
between <I>base pairs</I> of <I>nucleotides</I>. The four nucleotides in DNA
contain the bases: <I>adenine</I> (A), <I>guanine</I> (G), <I>cytosine</I>
(C), and <I>thymine</I> (T). In nature, <I>base pairs</I> form only between A
and T and between G and C; thus the <I>base sequence</I> of each single strand
can be deduced from that of its partner.</P>

<B><P>
<A NAME="dna probes">
DNA probes:</B> See <I>probe</I>.</P>

<B><P>
<A NAME="dna replication">
DNA replication:</B> The use of existing DNA as a template for the synthesis
of new DNA strands. In humans and other <I>eukaryotes</I>, replication occurs
in the cell <I>nucleus</I>.</P>

<B><P>
<A NAME="dna sequence">
DNA sequence:</B> The relative order of <I>base pairs</I>, whether in a
fragment of DNA, a <I>gene</I>, a <I>chromosome</I>, or an entire
<I>genome</I>. See <I>base sequence analysis</I>.</P>

<B><P>
<A NAME="domain">
Domain:</B> A discrete portion of a <I>protein</I> with its own function. The
combination of domains in a single protein determines its overall function.
</P>

<B><P>
<A NAME="double helix">
Double helix:</B> The shape that two linear strands of DNA assume when bonded
together.</P>

<B><P>
<A NAME="e. coli">
E. coli:</B> Common bacterium that has been studied intensively by geneticists
because of its small genome size, normal lack of pathogenicity, and ease of
growth in the laboratory.</P>

<B><P>
<A NAME="electrophoresis">
Electrophoresis:</B> A method of separating large molecules (such as DNA
fragments or <I>proteins</I>) from a mixture of similar molecules. An electric
current is passed through a medium containing the mixture, and each kind of
molecule travels through the medium at a different rate, depending on its
electrical charge and size. Separation is based on these differences. Agarose
and acrylamide gels are the media commonly used for electrophoresis of
proteins and nucleic acids.</P>

<B><P>
<A NAME="endonuclease">
Endonuclease:</B> An <I>enzyme</I> that cleaves its nucleic acid substrate at
internal sites in the <I>nucleotide</I> sequence.</P>

<B><P>
<A NAME="enzyme">
Enzyme:</B> A <I>protein</I> that acts as a catalyst, speeding the rate at
which a biochemical reaction proceeds but not altering the direction or nature
of the reaction.</P>

<B><P>
<A NAME="est">
EST:</B> Expressed sequence tag. See <I>sequence tagged site</I>.</P>

<B><P>
<A NAME="eukaryote">
Eukaryote:</B> Cell or organism with membrane-bound, structurally discrete
<I>nucleus</I> and other well-developed subcellular compartments. Eukaryotes
include all organisms except viruses, bacteria, and blue-green algae. Compare
<I>prokaryote</I>. See chromosomes.</P>

<B><P>
<A NAME="evolutionarily conserved">
Evolutionarily conserved:</B> See <I>conserved sequence</I>.</P>

<B><P>
<A NAME="exogenous dna">
Exogenous DNA:</B> DNA originating outside an organism.</P>

<B><P>
<A NAME="exons">
Exons:</B> The <I>protein</I>-coding DNA sequences of a <I>gene</I>. Compare
<I>introns</I>.</P>

<B><P>
<A NAME="exonuclease">
Exonuclease:</B> An <I>enzyme</I> that cleaves <I>nucleotides</I> sequentially
from free ends of a linear nucleic acid substrate.</P>

<B><P>
<A NAME="expressed gene">
Expressed gene:</B> See <I>gene expression</I>.</P>

<B><P>
<A NAME="fish">
<A NAME="fluorescence in situ hybridization">
FISH (fluorescence in situ hybridization):</B> A <I>physical mapping</I>
approach that uses fluorescein tags to detect <I>hybridization</I> of
<I>probes</I> with <I>metaphase chromosomes</I> and with the less-condensed
<I>somatic interphase</I> chromatin.</P>

<B><P>
<A NAME="flow cytometry">
Flow cytometry:</B> Analysis of biological material by detection of the light-
absorbing or fluorescing properties of cells or subcellular fractions (i.e.,
<I>chromosomes</I>) passing in a narrow stream through a laser beam. An
absorbance or fluorescence profile of the sample is produced. Automated
sorting devices, used to fractionate samples, sort successive droplets of the
analyzed stream into different fractions depending on the fluorescence emitted
by each droplet.</P>

<B><P>
<A NAME="flow karyotyping">
Flow karyotyping:</B> Use of flow cytometry to analyze and/or separate
<I>chromosomes</I> on the basis of their DNA content.</P>

<B><P>
<A NAME="gamete">
Gamete:</B> Mature male or female reproductive cell (sperm or ovum) with a
<I>haploid</I> set of <I>chromosomes</I> (23 for humans).</P>

<B><P>
<A NAME="gene">
Gene:</B> The fundamental physical and functional unit of heredity. A
<I>gene</I> is an ordered sequence of <I>nucleotides</I> located in a
particular position on a particular <I>chromosome</I> that encodes a specific
functional product (i.e., a <I>protein</I> or <I>RNA molecule</I>). See
<I>gene expression</I>.</P>

<B><P>
<A NAME="gene expression">
Gene expression:</B> The process by which a <I>genes</I> coded information is
converted into the structures present and operating in the cell. Expressed
genes include those that are transcribed into <I>mRNA</I> and then translated
into <I>protein</I> and those that are transcribed into <I>RNA</I> but not
translated into protein (e.g., <I>transfer</I> and <I>ribosomal RNAs</I>).</P>

<B><P>
<A NAME="gene families">
Gene families:</B> Groups of closely related <I>genes</I> that make similar
products.</P>

<B><P>
<A NAME="gene library">
Gene library:</B> See <I>genomic library</I>.</P>

<B><P>
<A NAME="gene mapping">
Gene mapping:</B> Determination of the relative positions of <I>genes</I> on a
DNA molecule (<I>chromosome</I> or <I>plasmid</I>) and of the distance, in
<I>linkage</I> units or physical units, between them.</P>

<B><P>
<A NAME="gene product">
Gene product:</B> The biochemical material, either <I>RNA</I> or
<I>protein</I>, resulting from expression of a gene. The amount of gene
product is used to measure how active a gene is; abnormal amounts can be
correlated with disease-causing alleles.</P>

<B><P>
<A NAME="genetic code">
Genetic code:</B> The sequence of <I>nucleotides</I>, coded in triplets
(<I>codons</I>) along the <I>mRNA</I>, that determines the sequence of
<I>amino acids</I> in <I>protein</I> synthesis. The DNA sequence of a
<I>gene</I> can be used to predict the mRNA sequence, and the genetic code can
in turn be used to predict the <I>amino acid</I> sequence.</P>

<B><P>
<A NAME="genetic engineering technologies">
Genetic engineering technologies:</B> See <I>recombinant DNA technologies</I>.
</P>

<B><P>
<A NAME="genetic map">
Genetic map:</B> See <I>linkage map</I>.</P>

<B><P>
<A NAME="genetic material">
Genetic material:</B> See <I>genome</I>.</P>

<B><P>
<A NAME="genetics">
Genetics:</B> The study of the patterns of inheritance of specific traits.</P>

<B><P>
<A NAME="genome">
Genome:</B> All the genetic material in the <I>chromosomes</I> of a particular
organism; its size is generally given as its total number of <I>base
pairs</I>.</P>

<B><P>
<A NAME="genome projects">
Genome projects:</B> Research and technology development efforts aimed at
<I>mapping</I> and <I>sequencing</I> some or all of the <I>genome</I> of human
beings and other organisms.</P>

<B><P>
<A NAME="genomic library">
Genomic library:</B> A collection of <I>clones</I> made from a set of randomly
generated overlapping DNA fragments representing the entire <I>genome</I> of
an organism. Compare <I>library, arrayed library</I>.</P>

<B><P>
<A NAME="global alignment">
Global Alignment:</B>Alignment of A within B (like agrep) Allows for gaps.</P>

<B><P>
<A NAME="guanine">
<A NAME="g">
Guanine (G):</B> A nitrogenous base, one member of the <I>base pair</I> G-C
(guanine and <I>cytosine</I>).</P>

<B><P>
<A NAME="haploid">
Haploid:</B> A single set of <I>chromosomes</I> (half the full set of genetic
material), present in the egg and sperm cells of animals and in the egg and
pollen cells of plants. Human beings have 23 chromosomes in their reproductive
cells. Compare <I>diploid</I>.</P>

<B><P>
<A NAME="heterozygosity">
Heterozygosity:</B> The presence of different <I>alleles</I> at one or more
<I>loci</I> on <I>homologous chromosomes</I>.</P>

<B><P>
<A NAME="homeobox">
Homeobox:</B> A short stretch of <I>nucleotides</I> whose <I>base sequence</I>
is virtually identical in all the <I>genes</I> that contain it. It has been
found in many organisms from fruit flies to human beings. In the fruit fly, a
homeobox appears to determine when particular groups of genes are expressed
during development.</P>

<B><P>
<A NAME="homologies">
Homologies:</B> Similarities in DNA or <I>protein</I> sequences between
individuals of the same species or among different species.</P>

<B><P>
<A NAME="homologous chromosomes">
Homologous chromosomes:</B> A pair of <I>chromosomes</I> containing the same
linear <I>gene</I> sequences, each derived from one parent.</P>

<B><P>
<A NAME="higest scoring paris">
<A NAME="hsp">
Highest Scoring Pairs(hsp):</B> The the highest scoring pair of alignments
between two <I>nucleotides</I> or <I>amino acids</I> sequences.</P>

<B><P>
<A NAME="human gene therapy">
Human gene therapy:</B> Insertion of normal DNA directly into cells to correct
a genetic defect. <P>

<B><P>
<A NAME="human genome initiative">
Human Genome Initiative:</B> Collective name for several projects begun in
1986 by DOE to (1) create an ordered set of DNA segments from known
chromosomal locations, (2) develop new computational methods for analyzing
genetic map and DNA sequence data, and (3) develop new techniques and
instruments for detecting and analyzing DNA. This DOE initiative is now known
as the Human Genome Program. The national effort, led by DOE and NIH, is known
as the Human Genome Project.</P>

<B><P>
<A NAME="hybridization">
Hybridization:</B> The process of joining two <I>complementary</I> strands of
DNA or one each of DNA and RNA to form a double-stranded molecule.</P>

<B><P>
<A NAME="indels">
InDels:</B> Insertion or Deletion. The insertion or deletion of an <I>amino
acid</I> or <I>nucleotide</I> from a protine or DNA sequence.</P>

<B><P>
<A NAME="informatics">
Informatics:</B> The study of the application of computer and statistical
techniques to the management of information. In <I>genome</I> projects,
informatics includes the development of methods to search databases quickly,
to analyze DNA sequence information, and to predict <I>protein</I> sequence
and structure from DNA sequence data.</P>

<B><P>
<A NAME="in situ hybridization">
In situ hybridization:</B> Use of a DNA or RNA probe to detect the presence of
the <I>complementary DNA</I> sequence in cloned bacterial or cultured
<I>eukaryotic</I> cells.</P>

<B><P>
<A NAME="interphase">
Interphase:</B> The period in the cell cycle when DNA is replicated in the
nucleus; followed by <I>mitosis</I>.</P>

<B><P>
<A NAME="introns">
Introns:</B> The DNA <I>base sequences</I> interrupting the <I>protein</I>-
coding sequences of a <I>gene</I>; these sequences are <I>transcribed</I> into
<I>RNA</I> but are cut out of the message before it is <I>translated</I> into
protein. Compare <I>exons</I>.</P>

<B><P>
<A NAME="in vitro">
In vitro:</B> Outside a living organism.</P>

<B><P>
<A NAME="karyotype">
Karyotype:</B> A photomicrograph of an individuals <I>chromosomes</I> arranged
in a standard format showing the number, size, and shape of each chromosome
type; used in low-resolution <I>physical mapping</I> to correlate gross
chromosomal abnormalities with the characteristics of specific diseases.</P>

<B><P>
<A NAME="kb">
kb:</B> See <I>kilobase</I>.</P>

<B><P>
<A NAME="kilobase">
<A NAME="kb">
Kilobase (kb):</B> Unit of length for DNA fragments equal to 1000
<I>nucleotides</I>.</P>

<B><P>
<A NAME="k-tup">
<A NAME="k-tuple">
k-tuple (k-tup):</B> a size &quot;k&quot; tuple of <I>nucleotides</I> or
<I>amino acids</I>.</P>

<B><P>
<A NAME="library">
Library:</B> An unordered collection of <I>clones</I> (i.e., cloned DNA from a
particular organism), whose relationship to each other can be established by
<I>physical mapping</I>. Compare <I>genomic library, arrayed library</I>.</P>

<B><P>
<A NAME="linkage">
Linkage:</B> The proximity of two or more <I>markers</I> (e.g., <I>genes,
RFLP</I> markers) on a <I>chromosome</I>; the closer together the markers are,
the lower the probability that they will be separated during DNA repair or
replication processes (binary fission in <I>prokaryotes, mitosis</I> or
<I>meiosis</I> in <I>eukaryotes</I>), and hence the greater the probability
that they will be inherited together.</P>

<B><P>
<A NAME="linkage map">
Linkage map:</B> A map of the relative positions of genetic <I>loci</I> on a
<I>chromosome</I>, determined on the basis of how often the loci are inherited
together. Distance is measured in <I>centimorgans (cM)</I>.</P>

<B><P>
<A NAME="local alignment">
Global Alignment:</B>Alignment of A to B like (diff) Looks for Longest Common
Subsequence (LCS).</P>

<B><P>
<A NAME="localize">
Localize:</B> Determination of the original position <I>(locus)</I> of a
<I>gene</I> or other <I>marker</I> on a chromosome.</P>

<B><P>
<A NAME="locus">
<A NAME="pl. loci">
Locus (pl. loci):</B> The position on a <I>chromosome</I> of a <I>gene</I> or
other chromosome <I>marker</I>; also, the DNA at that position. The use of
<I>locus</I> is sometimes restricted to mean regions of DNA that are
<I>expressed</I>. See <I>gene expression</I>.</P>

<B><P>
<A NAME="longest common subsequence">
<A NAME="lcs">
Longest Commmon Subequence (lcs):</B> The longest sequence of
<I>nucleotides</I> or <I>amino acids</I> common to two sequences.</P>

<B><P>
<A NAME="macrorestriction map">
Macrorestriction map:</B> Map depicting the order of and distance between sites
at which <I>restriction enzymes</I> cleave <I>chromosomes</I>.</P>

<B><P>
<A NAME="mapping">
Mapping:</B> See <I>gene mapping, linkage map, physical map</I>.</P>

<B><P>
<A NAME="marker">
Marker:</B> An identifiable physical location on a <I>chromosome</I> (e.g.,
<I>restriction enzyme cutting site, gene</I>) whose inheritance can be
monitored. Markers can be expressed regions of DNA (genes) or some segment of
DNA with no known coding function but whose pattern of inheritance can be
determined. See <I>RFLP, restriction fragment length polymorphism</I>.</P>

<B><P>
<A NAME="mb">
Mb:</B> See <I>megabase</I>.</P>

<B><P>
<A NAME="megabase">
<A NAME="mb">
Megabase (Mb):</B> Unit of length for DNA fragments equal to 1 million
<I>nucleotides</I> and roughly equal to 1 <I>cM</I>.</P>

<B><P>
<A NAME="meiosis">
Meiosis:</B> The process of two consecutive cell divisions in the
<I>diploid</I> progenitors of sex cells. Meiosis results in four rather than
two daughter cells, each with a <I>haploid</I> set of <I>chromosomes</I>.</P>

<B><P>
<A NAME="messenger rna">
<A NAME="mrna">
Messenger RNA (mRNA):</B> RNA that serves as a template for protein synthesis.
See genetic code.</P>

<B><P>
<A NAME="metaphase">
Metaphase:</B> A stage in <I>mitosis</I> or <I>meiosis</I> during which the
<I>chromosomes</I> are aligned along the equatorial plane of the cell.</P>

<B><P>
<A NAME="mitosis">
Mitosis:</B> The process of nuclear division in cells that produces daughter
cells that are genetically identical to each other and to the parent cell.</P>

<B><P>
<A NAME="mrna">
mRNA:</B> See <I>messenger RNA</I>.</P>

<B><P>
<A NAME="multifactorial or multigenic disorders">
Multifactorial or multigenic disorders:</B> See <I>polygenic disorders</I>.</P>

<B><P>
<A NAME="multiple alignment">
Multiple Alignment:</B> A <I>sequencing</I> approach that simultaneously aligns
<I>n</I> sequences.</P>

<B><P>
<A NAME="multiplexing">
Multiplexing:</B> A <I>sequencing</I> approach that uses several pooled
samples simultaneously, greatly increasing sequencing speed.</P>

<B><P>
<A NAME="mutation">
Mutation:</B> Any heritable change in DNA <I>sequence</I>. Compare
<I>polymorphism</I>.</P>

<B><P>
<A NAME="nitrogenous base">
Nitrogenous base:</B> A nitrogen-containing molecule having the chemical
properties of a base.</P>

<B><P>
<A NAME="nucleic acid">
Nucleic acid:</B> A large molecule composed of <I>nucleotide</I> subunits.</P>

<B><P>
<A NAME="nucleotide">
Nucleotide:</B> A subunit of DNA or RNA consisting of a nitrogenous base
(<I>adenine, guanine, thymine,</I> or <I>cytosine</I> in DNA; adenine, guanine,
<I>uracil</I>, or cytosine in RNA), a phosphate molecule, and a sugar molecule
(deoxyribose in DNA and ribose in RNA). Thousands of <I>nucleotides</I> are
linked to form a DNA or RNA molecule. See <I>DNA, base pair, RNA</I>.</P>

<B><P>
<A NAME="nucleus">
Nucleus:</B> The cellular organelle in <I>eukaryotes</I> that contains the
genetic material.</P>

<B><P>
<A NAME="oncogene">
Oncogene:</B> A <I>gene</I>, one or more forms of which is associated with
cancer. Many oncogenes are involved, directly or indirectly, in controlling the
rate of cell growth.</P>

<B><P>
<A NAME="overlapping clones">
Overlapping clones:</B> See <I>genomic library</I>.</P>

<B><P>
<A NAME="pam">
PAM:</B> Percent Accepted Mutation.  A group of weighted scoring matricies for
alignment of <I>nucleotied</I> or <I>DNA</I> sequences.</P>

<B><P>
<A NAME="pcr">
PCR:</B> See <I>polymerase chain reaction</I>.</P>

<B><P>
<A NAME="phage">
Phage:</B> A <I>virus</I> for which the natural host is a bacterial cell.</P>

<B><P>
<A NAME="physical map">
<A NAME="physical mapping">
Physical map:</B> A map of the locations of identifiable landmarks on DNA
(e.g., <I>restriction enzyme cutting sites, genes</I>), regardless of
inheritance. Distance is measured in <I>base pairs</I>. For the human
<I>genome</I>, the lowest-resolution <I>physical map</I> is the banding
patterns on the 24 different <I>chromosomes</I>; the highest-resolution map
would be the complete <I>nucleotide</I> sequence of the chromosomes.</P>

<B><P>
<A NAME="plasmid">
Plasmid:</B> Autonomously replicating, extrachromosomal circular DNA molecules,
distinct from the normal bacterial <I>genome</I> and nonessential for cell
survival under nonselective conditions. Some plasmids are capable of
integrating into the host genome. A number of artificially constructed plasmids
are used as <I>cloning vectors</I>.</P>

<B><P>
<A NAME="polygenic disorders">
Polygenic disorders:</B> Genetic disorders resulting from the combined action
of <I>alleles</I> of more than one <I>gene</I> (e.g., heart disease, diabetes,
and some cancers). Although such disorders are inherited, they depend on the
simultaneous presence of several alleles; thus the hereditary patterns are
usually more complex than those of <I>single-gene disorders</I>. Compare
single-gene disorders.</P>

<B><P>
<A NAME="polymerase chain reaction">
<A NAME="pcr">
Polymerase chain reaction (PCR):</B> A method for amplifying a DNA <I>base
sequence</I> using a heat-stable <I>polymerase</I> and two 20-base
<I>primers</I>, one <I>complementary</I> to the (+)-strand at one end of the
sequence to be amplified and the other complementary to the (-)-strand at the
other end. Because the newly synthesized DNA strands can subsequently serve as
additional templates for the same primer sequences, successive rounds of primer
annealing, strand elongation, and dissociation produce rapid and highly
specific amplification of the desired sequence. PCR also can be used to detect
the existence of the defined sequence in a DNA sample.</P>

<B><P>
<A NAME="polymerase">
Polymerase, DNA or RNA:</B> <I>Enzymes</I> that catalyze the synthesis of
<I>nucleic acids</I> on preexisting nucleic acid templates, assembling RNA from
ribonucleotides or DNA from deoxyribonucleotides.</P>

<B><P>
<A NAME="polymorphism">
Polymorphism:</B> Difference in DNA sequence among individuals. Genetic
variations occurring in more than 1% of a population would be considered useful
polymorphisms for genetic <I>linkage</I> analysis. Compare <I>mutation</I>.</P>

<B><P>
<A NAME="primer">
Primer:</B> Short preexisting polynucleotide chain to which new
deoxyribonucleotides can be added by DNA <I>polymerase</I>.</P>

<B><P>
<A NAME="probe">
Probe:</B> Single-stranded DNA or RNA molecules of specific base
<I>sequence</I>, labeled either radioactively or immunologically, that are used
to detect the <I>complementary</I> base sequence by <I>hybridization</I>.</P>

<B><P>
<A NAME="prokaryote">
Prokaryote:</B> Cell or organism lacking a membrane-bound, structurally
discrete <I>nucleus</I> and other subcellular compartments. Bacteria are
prokaryotes. Compare <I>eukaryote</I>. See <I>chromosomes</I>.</P>

<B><P>
<A NAME="promoter">
Promoter:</B> A site on DNA to which <I>RNA polymerase</I> will bind and
initiate <I>transcription</I>.</P>

<B><P>
<A NAME="protein">
Protein:</B> A large molecule composed of one or more chains of <I>amino
acids</I> in a specific order; the order is determined by the <I>base
sequence</I> of <I>nucleotides</I> in the <I>gene</I> coding for the protein.
Proteins are required for the structure, function, and regulation of the bodys
cells, tissues, and organs, and each protein has unique functions. Examples are
hormones, <I>enzymes</I>, and antibodies.</P>

<B><P>
<A NAME="purine">
Purine:</B> A nitrogen-containing, single-ring, basic compound that occurs in
nucleic acids. The purines in DNA and RNA are adenine and guanine.</P>

<B><P>
<A NAME="pyrimidine">
Pyrimidine:</B> A nitrogen-containing, double-ring, basic compound that occurs
in nucleic acids. The pyrimidines in DNA are cytosine and thymine; in RNA,
cytosine and uracil.</P>

<B><P>
<A NAME="cutter enzyme">
Rare-cutter enzyme:</B> See <I>restriction enzyme cutting site</I>.</P>

<B><P>
<A NAME="recombinant clones">
Recombinant clones:</B> <I>Clones</I> containing <I>recombinant DNA
molecules</I>. See <I>recombinant DNA technologies</I>.</P>

<B><P>
<A NAME="recombinant clones">
Recombinant DNA molecules:</B> A combination of DNA molecules of different
origin that are joined using <I>recombinant DNA technologies</I>.</P>

<B><P>
<A NAME="recombinant dna technologies">
Recombinant DNA technologies:</B> Procedures used to join together DNA
segments in a cell-free system (an environment outside a cell or organism).
Under appropriate conditions, a recombinant DNA molecule can enter a cell and
replicate there, either autonomously or after it has become integrated into a
cellular <I>chromosome</I>.</P>

<B><P>
<A NAME="recombination">
Recombination:</B> The process by which progeny derive a combination of
<I>genes</I> different from that of either parent. In higher organisms, this
can occur by <I>crossing over</I>.</P>

<B><P>
<A NAME="regulatory regions or sequences">
Regulatory regions or sequences:</B> A DNA <I>base sequence</I> that controls
<I>gene expression</I>.</P>

<B><P>
<A NAME="resolution">
Resolution:</B> Degree of molecular detail on a <I>physical map</I> of DNA,
ranging from low to high.</P>

<B><P>
<A NAME="restriction enzyme">
<A NAME="endonuclease">
Restriction enzyme, endonuclease:</B> A <I>protein</I> that recognizes
specific, short <I>nucleotide sequences</I> and cuts DNA at those sites.
Bacteria contain over 400 such <I>enzymes</I> that recognize and cut over 100
different DNA sequences. See <I>restriction enzyme cutting site</I>.</P>

<B><P>
<A NAME="restriction enzyme cutting site">
Restriction enzyme cutting site:</B> A specific <I>nucleotide sequence</I> of
DNA at which a particular <I>restriction enzyme</I> cuts the DNA. Some
sitesoccur frequently in DNA (e.g., every several hundred <I>base pairs</I>),
others much less frequently (<I>rare-cutter</I>; e.g., every 10,000 base
pairs).
</P>

<B><P>
<A NAME="restriction fragment length polymorphism">
<A NAME="rflp">
Restriction fragment length polymorphism (RFLP):</B> Variation between
individuals in DNA fragment sizes cut by specific <I>restriction enzymes;
polymorphic sequences</I> that result in RFLPs are used as <I>markers</I>
on both <I>physical maps</I> and genetic <I>linkage maps</I>. RFLPs are
usually caused by <I>mutation</I> at a cutting site. See <I>marker</I>.</P>

<B><P>
<A NAME="rflp">
RFLP:</B> See <I>restriction fragment length polymorphism</I>.</P>

<B><P>
<A NAME="ribonucleic acid">
<A NAME="rna">
Ribonucleic acid (RNA):</B> A chemical found in the <I>nucleus</I> and
cytoplasm of cells; it plays an important role in protein synthesis and other
chemical activities of the cell. The structure of RNA is similar to that of
DNA. There are several classes of RNA molecules, including <I>messenger RNA,
transfer RNA, ribosomal RNA</I>, and other small RNAs, each serving a different
purpose.</P>

<B><P>
<A NAME="ribonucleotides">
Ribonucleotides:</B> See <I>nucleotide</I>.</P>

<B><P>
<A NAME="ribosomal rna">
<A NAME="rrna">
Ribosomal RNA (rRNA):</B> A class of RNA found in the ribosomes of cells.</P>

<B><P>
<A NAME="ribosomes">
Ribosomes:</B> Small cellular components composed of specialized ribosomal RNA
and protein; site of protein synthesis. See <I>ribonucleic acid (RNA)</I>.</P>

<B><P>
<A NAME="rna">
RNA:</B> See <I>ribonucleic acid</I>.</P>

<B><P>
<A NAME="sequence">
Sequence:</B> See <I>base sequence</I>.</P>

<B><P>
<A NAME="sequence tagged site">
<A NAME="sts">
Sequence tagged site (STS):</B> Short (200 to 500 <I>base pairs</I>) DNA
sequence that has a single occurrence in the human genome and whose location
and base sequence are known. Detectable by <I>polymerase chain reaction</I>,
STSs are useful for localizing and orienting the mapping and sequence data
reported from many different laboratories and serve as landmarks on the
developing <I>physical map</I> of the human genome. Expressed sequence tags
(ESTs) are STSs derived from cDNAs.</P>

<B><P>
<A NAME="sequencing">
Sequencing:</B> Determination of the order of <I>nucleotides (base
sequences)</I> in a DNA or <I>RNA</I> molecule or the order of <I>amino
acids</I> in a <I>protein</I>.</P>

<B><P>
<A NAME="sex chromosomes">
Sex chromosomes:</B> The X and Y <I>chromosomes</I> in human beings that
determine the sex of an individual. Females have two X chromosomes in diploid
cells; males have an X and a Y chromosome. The sex chromosomes comprise the
23rd chromosome pair in a <I>karyotype</I>. Compare <I>autosome</I>.</P>

<B><P>
<A NAME="shotgun method">
Shotgun method:</B> <I>Cloning</I> of DNA fragments randomly generated from a
<I>genome</I>. See <I>library, genomic library</I>.</P>

<B><P>
<A NAME="gene disorder">
Single-gene disorder:</B> Hereditary disorder caused by a <I>mutant</I> allele
of a single <I>gene</I> (e.g., Duchenne muscular dystrophy, retinoblastoma,
sickle cell disease). Compare <I>polygenic disorders</I>.</P>

<B><P>
<A NAME="somatic cells">
Somatic cells:</B> Any cell in the body except <I>gametes</I> and their
precursors.</P>

<B><P>
<A NAME="southern blotting">
Southern blotting:</B> Transfer by absorption of DNA fragments separated in
electrophoretic gels to membrane filters for detection of specific <I>base
sequences</I> by radiolabeled complementary probes.</P>

<B><P>
<A NAME="sts">
STS:</B> See <I>sequence tagged site</I>.</P>

<B><P>
<A NAME="tandem repeat sequences">
Tandem repeat sequences:</B> Multiple copies of the same <I>base sequence</I>
on a <I>chromosome</I>; used as a marker in <I>physical mapping</I>.</P>

<B><P>
<A NAME="technology transfer">
Technology transfer:</B> The process of converting scientific findings from
research laboratories into useful products by the commercial sector.</P>

<B><P>
<A NAME="telomere">
Telomere:</B> The ends of <I>chromosomes</I>. These specialized structures are
involved in the replication and stability of linear DNA molecules. See <I>DNA
replication</I>.</P>

<B><P>
<A NAME="thymine">
<A NAME="t">
Thymine (T):</B> A nitrogenous base, one member of the <I>base pair</I> A-T
(<I>adenine</I>-thymine).</P>

<B><P>
<A NAME="transcription">
Transcription:</B> The synthesis of an <I>RNA</I> copy from a <I>sequence</I>
of DNA (a <I>gene</I>); the first step in <I>gene expression</I>. Compare
<I>translation</I>.</P>

<B><P>
<A NAME="transfer rna">
<A NAME="trna">
Transfer RNA (tRNA):</B> A class of <I>RNA</I> having structures with triplet
<I>nucleotide</I> sequences that are <I>complementary</I> to the triplet
nucleotide coding sequences of <I>mRNA</I>. The role of tRNAs in protein
synthesis is to bond with <I>amino acids</I> and transfer them to the
ribosomes, where proteins are assembled according to the genetic code carried
by mRNA.</P>

<B><P>
<A NAME="transformation">
Transformation:</B> A process by which the genetic material carried by an
individual cell is altered by incorporation of exogenous DNA into its
<I>genome</I>.</P>

<B><P>
<A NAME="transition">
Transition:</B> The term proposed by Freese (1959) for a mutation caused by the
substitution in DNA or RNA of one <I>purine</I> by the other, and similarly
with the <I>pyrimidines</I>. (see <I>transversion</I>)</P>

<B><P>
<A NAME="translation">
Translation:</B> The process in which the genetic code carried by mRNA directs
the synthesis of <I>proteins</I> from amino acids. Compare
<I>transcription</I>.</P>

<B><P>
<A NAME="transversion">
Transversion:</B> The term proposed by Freese (1959) for a mutation caused by
the substitution of a <I>purine</I> for a <I>pyrimidine</I>, and vice versa,
in DNA or RNA. (see <I>Transition</I>).</P>

<B><P>
<A NAME="trna">
tRNA:</B> See <I>transfer RNA</I>.</P>

<B><P>
<A NAME="uracil">
Uracil:</B> A nitrogenous base normally found in RNA but not DNA; uracil is
capable of forming a <I>base pair</I> with <I>adenine</I>.</P>

<B><P>
<A NAME="vector">
Vector:</B> See <I>cloning vector</I>.</P>

<B><P>
<A NAME="virus">
Virus:</B> A noncellular biological entity that can reproduce only within a
host cell. Viruses consist of <I>nucleic acid</I> covered by <I>protein</I>;
some animal viruses are also surrounded by membrane. Inside the infected cell,
the virus uses the synthetic capability of the host to produce progeny
virus.</P>

<B><P>
<A NAME="vlsi">
VLSI:</B> Very large-scale integration allowing over 100,000 transistors on a
chip.</P>

<B><P>
<A NAME="yac">
YAC:</B> See <I>yeast artificial chromosome</I>. <P>

<B><P>
<A NAME="yeast artificial chromosome">
<A NAME="yac">
Yeast artificial chromosome (YAC):</B> A vector used to clone DNA fragments (up
to 400 kb); it is constructed from the telomeric, centromeric, and replication
origin sequences needed for replication in yeast cells. Compare <I>cloning
vector, cosmid</I>.</P>

</BODY>
</HTML>